### DISCUSSION TITLE
Thoughts on Future Developments

### DISCUSSION THREAD

**Discussion by chaosarium** [HugoFara/lwt#14] [General]
Hi! It looks like there's been lots of work going into LWT in the past few months. I really enjoy LWT's approach and will try to contribute in any way I can.

So far, I have noticed some problems in the newest version (2.2.0). I also have a bunch of ideas in mind about LWT that I think might be helpful for the community to consider.

## Problems when using current version

I have noticed saveral problems with the current version that still make me hesitate to migrate from my fork. I opened a new issue for each of them. Here's a summary:

- #8
- #9
- #10
- #11
- #12

## Questions about user experience

- **Location of playback control?** - As far as I'm aware, the current version puts the playback control at the bottom, so controlling playback requires scrolling all the way to the bottom, which I feel is less efficient than the old LWT with sticky playback control. I wonder if this is an intended feature or if there should be a setting to control where the playback control is.
- **Sticky panels?** - Sometimes the lookup panel slides in and blocks content. I wonder if there should be a way to make panels sticky for things like heavy lookups, for example. Perhaps sticky panels like those code editors might help. 

## Feature ideas

- **Text to speech for individual words.** I've already worked on this a little bit on my branch (commit https://github.com/chaosarium/lwt/commit/8637e92a593532d7a68e6dc150deeda9bef57e6e). I don't think it's the most elegant solution but it's a proof of concept
- **Offline dictionary?**. I have experimented with running a local server that serves a .mdx dictionary, but it didn't work quite well, especially because lwt treats each inflection of a lemma separately. (for example, looking up "spoke" doesn't work because the lemma is "speak")
- **Add setting to toggle status chart scale**. I hard-coded the log-scale charts in commit https://github.com/chaosarium/lwt/commit/bed0689446d5520b350d37c267fd115c71371af9 for my personal use, but this might not be everyone's cup of tea.
- **Allow icon themes?**. I feel like this might be necessary if LWT wants to allow more theme customisation or update its interface to a more modern look.
- **Lemma possible?** A language tool similar to LWT, https://www.vocabtracker.com, has an option to use lemma
    > Use Variant: words like "took", "taking", "taken" will be treated as different words.
    > Use Lemma: words like "took","taking","taken" will be treated as one word, which is their lemma "take".
- **Double click on word to jump to video location.** I enjoyed the ability to jump to the approximate time in audio when doubled clicked on a word in the text. I think it helps to have this feature for videos too.
- **Resize video player?** Right now videos seem squashed and hard to see.
- **More theming?** I feel like LWT's interface could look much better and there could be more themes. I do front-end development so this might be something I can help. This isn't the top priority right now though.

**Comment by HugoFara** [comment]
Hello!

Thank you for your interest! I will try to get through your main points of concern.

## Bugs
Well, we will address them in their respective threads. Thanks for signaling them!

## User Experience
* **Location of playback control**. That's ... a bug. I may have moved it unintentionally, this will get fixed as soon as possible!
* **Sticky panels**. What kind of panel are you talking about? The iframes on the right when reading (containing dictionaries for instance)?

## Ideas
Just create issues/discussions when you have precise ideas, it will help us track progress :)

* **Video player**. I had to find an equilibrium between the size of the video, and space used on screen. I considered that the video itself was not very useful since users should be reading the text. I may make it a little bigger though.
* **More theming**. Indeed, is the interface has an "old" aspect: lots of little icons. It may need an important refactor. For now I want to focus more on "making the interface customizable", so that the community would be able to go further with themes!
